src/hotspot/os/windows/os_windows.cpp

```diff
@@ -4090,6 +4091,33 @@ int    os::win32::_build_minor               = 0;
 bool   os::win32::_processor_group_warning_displayed = false;
 bool   os::win32::_job_object_processor_group_warning_displayed = false;
 
+std::string GetWindowsInstallationType() {
+  HKEY hKey;
+  const char* subKey = "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion";
+  const char* valueName = "InstallationType";
+  char value[256];
+  DWORD valueLength = sizeof(value);
+
+  // Open the registry key
+  if (RegOpenKeyExA(HKEY_LOCAL_MACHINE, subKey, 0, KEY_READ, &hKey) != ERROR_SUCCESS) {
+      return "";
+  }
+
+  // Query the value
+  if (RegQueryValueExA(hKey, valueName, NULL, NULL, (LPBYTE)value, &valueLength) != ERROR_SUCCESS) {
+      RegCloseKey(hKey);
+      return "";
+  }
+
+  RegCloseKey(hKey);
+  return std::string(value, valueLength - 1); // Exclude null terminator
+}
+
+bool IsNanoServer() {
+  std::string installationType = GetWindowsInstallationType();
+  return installationType == "Nano Server";
```

Comment:
Not sure how OpenJDK source code typically deals with magic strings, but should this get extracted to a const?